lib/storage/src/content_manager/migration.rs

```diff
@@ -0,0 +1,172 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use collection::collection::Collection;
+use collection::operations::point_ops::{PointInsertOperations, PointOperations, PointStruct};
+use collection::operations::types::{CollectionError, CollectionResult, ScrollRequest};
+use collection::operations::CollectionUpdateOperations;
+use collection::shards::replica_set::ReplicaState;
+use collection::shards::shard::{PeerId, ShardId};
+use collection::shards::CollectionId;
+use segment::types::{WithPayloadInterface, WithVector};
+use tokio::sync::RwLock;
+
+use crate::content_manager::collections_ops::Collections;
+
+const MIGRATION_BATCH_SIZE: usize = 1000;
+const COLLECTION_INITIATION_TIMEOUT: Duration = Duration::from_secs(60);
+
+/// Handlers for migration data from one collection into another within single cluster
+
+/// Get a list of local shards, which can be used for migration
```

Comment:
I find the usage of the term `migration` in this file and this context a bit confusing.
As far as I can see, it is actually a copy and not a migration as the source is not deleted at the end of the routine.

Source and target coexist at the end of the process.

WDYT?

Reply:
renames, not sure it became better, though

Reply:
Thanks for taking the time, I like it better. It is internal stuff anyway, we can rename it later :+1: